src/Workspaces/Core/Desktop/Workspace/Host/TemporaryStorage/TemporaryStorageServiceFactory.CopiedMemoryMappedViewHandle.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Microsoft.Win32.SafeHandles;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    internal partial class TemporaryStorageServiceFactory
+    {
+        /// <summary>
+        /// This critical handle increments the reference count of a <see cref="SafeMemoryMappedViewHandle"/>. The view
+        /// will be released when it is disposed <em>and</em> all pointers acquired through
+        /// <see cref="SafeBuffer.AcquirePointer"/> (which this class uses) are released.
+        /// </summary>
+        /// <remarks>
+        /// <para><see cref="CriticalHandle"/> types are not reference counted, and are thus somewhat limited in their
+        /// usefulness. However, this handle class has tightly restricted accessibility and is only used by managed
+        /// code which does not rely on it counting references.</para>
+        /// </remarks>
+        private unsafe sealed class CopiedMemoryMappedViewHandle : CriticalHandleZeroOrMinusOneIsInvalid
+        {
```

Comment:
What finalizer?  I don't see one in the current code.  

> This change would totally break debug builds.

If that's the case, can you comment as such that that's the intended way for this type to be used.   i.e. that the client of it is not required to release it since a finalizer will take care of it for them.

Again, this is non-obvious stuff.  We need documentation to explain the intent here and how its expected to be used properly.